### PR TITLE
deploy: default maxSurge/maxUnavailable separately

### PR DESCRIPTION
**Pull request by kargakis**
Restores defaulting for maxSurge/maxUnavailable (regressed in https://github.com/openshift/origin/pull/11090)

@mfojtik ptal

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1389170

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/11649 and https://github.com/openshift/origin/issues/10988 [test]

File: pkg/deploy/controller/deploymentconfig/controller.go [hunk 1]
```diff
@@ -351,7 +351,7 @@ func updateConditions(config deployapi.DeploymentConfig, newStatus *deployapi.De
 	if latestRC != nil {
 		switch deployutil.DeploymentStatusFor(latestRC) {
 		case deployapi.DeploymentStatusPending:
-			msg := fmt.Sprintf("Waiting on deployer pod for replication controller %q to be scheduled", latestRC.Name)
+			msg := fmt.Sprintf("Waiting deployer pod for replication controller %q to run", latestRC.Name)
```
Comment (mfojtik) [inline pkg/deploy/controller/deploymentconfig/controller.go]: Deployer pod waiting for replication controller :-)

**Comment by kargakis** [comment]
@mfojtik updated, ptal

**Comment by mfojtik** [comment]
LGTM [merge]

This is fixing BZ 1389170.

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/10228 [test]

**Comment by kargakis** [comment]
[test]

**Pull request merged**